src/main/java/com/uber/cadence/internal/common/InternalUtils.java

```diff
@@ -193,7 +193,8 @@ public static History DeserializeFromBlobToHistoryEvents(
       try {
         blob.setData(serializer.serialize(event));
       } catch (org.apache.thrift.TException err) {
-        throw new EntityNotExistsError("Deserialize blob data to history event failed with unknown error");
+        throw new EntityNotExistsError(
+            "Deserialize blob data to history event failed with unknown error");
```

Comment:
While you are touching this line, consider appending err.getMessage() to this error message for visibility.

Reply:
This error type is wrong. I've fixed it.